ci-scripts/JenkinsFile-GitLab

```diff
@@ -80,45 +77,30 @@ pipeline {
     stage ("Retrieve and Prepare Source Code") {
       steps {
         script {
-          echo "Git URL         is ${GIT_URL}"
-          echo "GitLab Act      is ${env.gitlabActionType}"
-          sh "git clean -x -d -e .cache -e lte/gateway/.vagrant -f > /dev/null 2>&1"
-          if ("MERGE".equals(env.gitlabActionType)) {
-            // since a bit, in push events, gitlabUserEmail is not populated
-            gitCommitAuthorEmailAddr = env.gitlabUserEmail
-            echo "GitLab Usermail is ${gitCommitAuthorEmailAddr}"
-            // GitLab-Jenkins plugin integration is lacking to perform the merge by itself
-            // Doing it manually --> it may have merge conflicts
-            gitlabCommitStatus(name: "Mergeability") {
-              sh "./ci-scripts/doGitLabMerge.sh --src-branch ${env.gitlabSourceBranch} --src-commit ${env.gitlabMergeRequestLastCommit} --target-branch ${env.gitlabTargetBranch} --target-commit ${GIT_COMMIT}"
-            }
-          } else {
-            echo "Git Branch      is ${GIT_BRANCH}"
-            echo "Git Commit      is ${GIT_COMMIT}"
-            // since a bit, in push events, gitlabUserEmail is not populated
-            gitCommitAuthorEmailAddr = sh returnStdout: true, script: 'git log -n1 --pretty=format:%ae ${GIT_COMMIT}'
-            gitCommitAuthorEmailAddr = gitCommitAuthorEmailAddr.trim()
-            echo "GitLab Usermail is ${gitCommitAuthorEmailAddr}"
-            sh "git log -n1 --pretty=format:\"%s\" > .git/CI_COMMIT_MSG"
-          }
-          TEMP_COMMIT = sh returnStdout: true, script: 'git log -n1 --pretty=format:"%H"'
-          TEMP_COMMIT = TEMP_COMMIT.trim()
-
-          sh "tar -cjhf /tmp/converged_mme.tar.bz2 .git"
-          sh "mv /tmp/converged_mme.tar.bz2 ."
-          copyTo2ndServer('converged_mme.tar.bz2', '.', new_host_flag, new_host_user, new_host)
-          myShCmd('git checkout -f ' + TEMP_COMMIT, new_host_flag, new_host_user, new_host)
-          myShCmd('git clean -x -d -e .cache -e lte/gateway/.vagrant -f > /dev/null 2>&1', new_host_flag, new_host_user, new_host)
-          myShCmd('git status --ignored', new_host_flag, new_host_user, new_host)
-          sh "mkdir -p archives"
-          myShCmd('mkdir -p archives', new_host_flag, new_host_user, new_host)
+            checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '${sha1}']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'vik_github', refspec: '+refs/pull/*:refs/remotes/origin/pr/*', url: "https://{{GIT_URL}}.git"]]]    
```

Comment:
@119Vik 

2 things about this line.

-  Now that we have a public repository, do we still need to provide credentials? I don't think so.
-  I am not sure that you are handling the pull request correctly: my meaning is that we shall test the potential merge of the contributor's branch with the target branch. Not just the contributor's branch.
   *  I understand that the policy is to force the contributor to resync its branch against the latest version of  `master`.
   *  But then you need  a check to ensure it
   *  That is what I was doing in the section you removed.

Reply:
Both make sense for me - let me update my PR.